src/full/Agda/Syntax/Translation/ConcreteToAbstract.hs

```diff
@@ -1979,14 +1984,14 @@ instance ToAbstract NicePatternSyn [A.Declaration] where
              lhs <- toAbstract (C.patternToExpr pat)
              let A.Application e as = A.appView lhs
              case e of
-               A.PatternSyn (AmbQ (y' :| [])) | y == y' ->
+               A.PatternSyn (AmbQ ys) | any (y ==) ys -> -- We are allowed to overload pattern synonyms!
```

Comment:
Remind me: what does `ys` mean here? And do we still avoid overloading of untyped pattern synonyms?

Reply:
It's a list of abstract names that can all be resolved from the same concrete name because e.g. overloading

Reply:
Why do we need to care about that at definition time?

Reply:
Because we are trying to parse the LHS and we need to make sure that we
got a parse that does correspond to the pattern synonym we are defining.
Otherwise the user may need to add parens / change the fixities involved.

Reply:
Aah, and this name is resolving as we view it. Makes sense. What happens when we try defining a name-clashing *untyped* pattern synonym, then?

Reply:
Untyped ones have a really easy to parse LHS. The binding comes later (and was already working before our changes). We will need to write test cases for all of these possible overlappings if course but this is a step towards that is at least backwards compatible